File: Java/07_Linked List & Arrays/E485.Max consecutive ones.java
// Max consecutive ones
// https://leetcode.com/problems/max-consecutive-ones/
// Time Complexity: O(n) - Single pass through array
// Space Complexity: O(1) - Only using constant extra space

class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int count = 0;
        for (int num : nums) {
            if (num == 1) {
                count++;
                max = Math.max(max, count);
            } else {
                count = 0;
            }
        }
        return max;
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {1, 1, 0, 1, 1, 1};
        int result = solution.findMaxConsecutiveOnes(nums);
        System.out.println("Max consecutive ones: " + result);
    }
}

File: JavaScript/07_Linked List & Arrays/E485.Max consecutive ones.js
// 485. Max Consecutive Ones
// https://leetcode.com/problems/max-consecutive-ones/
// Time Complexity: O(n) - Single pass through array
// Space Complexity: O(1) - Only using constant extra space

/**
 * @param {number[]} nums
 * @return {number}
 */
function findMaxConsecutiveOnes(nums) {
    let max = 0;
    let count = 0;

    for (const num of nums) {
        if (num === 1) {
            count++;
            max = Math.max(max, count);
        } else {
            count = 0;
        }
    }

    return max;
}

const nums = [1, 1, 0, 1, 1, 1];
console.log("Max consecutive ones:", findMaxConsecutiveOnes(nums)); // Output: 3
